### PR TITLE
Clean up SignHandler

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

- Deprecate SignHandler, turn it into set of util functions.
- Provider now manages the signer, reducing state sync overhead
- Pull out popup interaction logic from WLSigner

### _How did you test your changes?_

test cases for SignHandler have been updated and moved to Provider tests.

### CODE REVIEW

File: packages/wallet-sdk/src/core/message/ConfigMessage.ts [hunk 1]
```diff
@@ -11,4 +11,4 @@ export type ConfigEvent =
   | 'WalletLinkSessionRequest'
   | 'WalletLinkUpdate';
 
-export type SignerType = 'scw' | 'walletlink' | 'extension';
+export type SignerType = 'scw' | 'walletlink';
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/message/ConfigMessage.ts:14]: add it back when we start to support

File: packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts [hunk 2]
```diff
@@ -30,14 +25,84 @@ describe('CoinbaseWalletProvider', () => {
...
+    SCWSigner: jest.fn().mockImplementation(() => {
+      return {
+        handshake: mockHandshake,
+        request: mockRequest,
+      };
+    }),
+  };
+});
+
+describe('signer configuration', () => {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts:60]: moved from packages/wallet-sdk/src/sign/SignHandler.test.ts

File: packages/wallet-sdk/src/sign/SignHandler.ts [hunk 3]
```diff
@@ -1,109 +1,61 @@
...
-      walletlink: WLSigner,
-      extension: undefined,
-    };
-    return new SignerClasses[signerType]!({
-      metadata: this.metadata,
-      postMessageToPopup: this.communicator.postMessage.bind(this),
-      updateListener: this.listener,
-    });
-  }
+async function listenForWalletLinkSessionRequest(
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/SignHandler.ts:35]: moved from WLSigner

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 4]
```diff
@@ -135,4 +156,28 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
...
+      case 'scw':
+        return new SCWSigner({
+          metadata: this.metadata,
+          updateListener: this.updateListener,
+          postMessageToPopup: this.communicator.postMessage.bind(this),
+        });
+      case 'walletlink':
+        return new WLSigner({
+          metadata: this.metadata,
+          updateListener: this.updateListener,
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:179]: gonna revisit this. there should be a way to avoid `updateListener`

**Review by fan-zhang-sv — APPROVED** [review]
tested locally, work as expected

**Pull request merged**